portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,74 +14,80 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory; 
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.io.PrintWriter;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.sql.Timestamp;
+import java.util.Arrays;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Arrays;
-import java.sql.Timestamp;
-import com.google.gson.Gson;
-import com.google.appengine.api.datastore.PreparedQuery;
-import com.google.appengine.api.datastore.Query;
-import com.google.appengine.api.datastore.Query.SortDirection;
-import com.google.appengine.api.datastore.DatastoreService;
-import com.google.appengine.api.datastore.DatastoreServiceFactory;
-import com.google.appengine.api.datastore.Entity;
-import com.google.appengine.api.datastore.Key;
-import com.google.appengine.api.datastore.KeyFactory; 
-import com.google.appengine.api.datastore.FetchOptions;
-
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    private final int defaultMaxComments = 15;
+
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+        int max_comments;
+        if (request.getParameter("maxComments") == null) {
+            response.sendError(response.SC_BAD_REQUEST, "maxComment parameter missing");
+            max_comments = defaultMaxComments;
+        }
 
-        Query query = new Query("Task");
+        try{
+            max_comments = Integer.parseInt(request.getParameter("maxComments"));
+        } catch(NumberFormatException e) {
+            response.sendError(response.SC_BAD_REQUEST, "Value entered in \'Set maximum number of comment\' is not a number!");
+            max_comments = defaultMaxComments;
```

Comment:
After responding with an error you should not continue with a default value.

Reply:
Oops, will correct it!